_books/ion-1-1/src/macros/macros_by_example.md

```diff
@@ -529,90 +525,83 @@ Since the scale accepts the empty stream, we can pass it an empty argument group
 ```
 
 Note that the result’s `scale` field has disappeared because no value was provided. It would be
-more useful to fill in a default value, and to do that we introduce a special form that can
-detect the empty stream:
+more useful to fill in a default value, which we can achieve with the `default` system macro:
 
 ```ion
-(macro temperature
-  (decimal::degrees symbol::scale?)
-  {degrees: degrees, scale: (if_none scale (literal K) scale)})
+(macro temperature (degrees scale?)
+  {
+    degrees: (%degrees),
+    scale: (.default (%scale) K)
+  })
 ```
 ```ion
 (:temperature 96 F)    ⇒ {degrees:96,  scale:F}
 (:temperature 283 (:)) ⇒ {degrees:283, scale:K}
 ```
 
-The `if_none` form is if/then/else syntax testing stream emptiness. It has three sub-expressions,
-the first being a stream to check. If and only if that stream is empty (it produces no
-values), the second sub-expression is expanded and its results are returned by the `if_none`
-expression. Otherwise, the third sub-expression is expanded and returned.
-
-> [!NOTE]
-> Exactly one branch is expanded, because otherwise the empty stream might be used in a context
-> that requires a value, resulting in an errant expansion error.
-
 To refine things a bit further, trailing arguments that accept the empty stream can be omitted entirely:
 
 ```ion
 (:temperature 283) ⇒ {degrees:283, scale:K}
 ```
 
+> [!TIP]
+> The `default` macro is implemented with the help of a special form that can detect the empty stream: [`if_none`](special_forms.md#if_none).
 
 #### Zero-or-More
 
-A parameter with the modifier `*` has _zero-or-more cardinality_.
+A parameter with the modifier `*` has _`zero-or-more` cardinality_.
 
 ```ion
-(macro prices
-  (number::amount* symbol::currency)
-  (for [(amt amount)]
-    (price amt currency)))
+(macro prices (amount* currency)
+  (.for [(amt (%amount))]
+    (.price (%amt) (%currency))))
 ```
 
-When `*` is on a non-final parameter, we cannot take the “all the rest” of the arguments
+When `*` is on a non-final parameter, we cannot take “all the rest” of the arguments
 and must use a different calling convention to draw the boundaries of the stream.
 Instead, we need a single
 expression that produces the desired values:
 
 ```ion
-(:prices (:) JPY)          ⇒ _void_
+(:prices (:) JPY)          ⇒ // empty stream
 (:prices 54 CAD)           ⇒ {amount:54, currency:CAD}
 (:prices (: 10 9.99) GBP)  ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
```

Comment:
```suggestion
(:prices (::) JPY)          ⇒ // empty stream
(:prices 54 CAD)           ⇒ {amount:54, currency:CAD}
(:prices (:: 10 9.99) GBP)  ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
```
I think we need expression group syntax for these. These examples recur below too.